pkg/controller/deadmanssnitch/deadmanssnitch_controller.go

```diff
@@ -155,15 +155,47 @@ func (r *ReconcileDeadMansSnitch) Reconcile(request reconcile.Request) (reconcil
 	}
 
 	ssName := request.Name + config.SyncSetPostfix
+	secretName := request.Name + config.SecretPostfix
+
 	// Check to see if the SyncSet exists
 	err = r.client.Get(context.TODO(),
 		types.NamespacedName{Name: ssName, Namespace: request.Namespace},
 		&hivev1alpha1.SyncSet{})
+
 	if errors.IsNotFound(err) {
 		// create new DMS SyncSet
 		reqLogger.Info("SyncSet not found, Creating a new SynsSet", "Namespace", request.Namespace, "Name", request.Name)
 
-		snitchName := instance.Spec.ClusterName + "." + instance.Spec.BaseDomain
+		newSS := newSyncSet(request.Namespace, request.Name)
+
+		// ensure the syncset gets cleaned up when the clusterdeployment is deleted
+		if err := controllerutil.SetControllerReference(instance, newSS, r.scheme); err != nil {
+			reqLogger.Error(err, "Error setting controller reference on syncset", "Namespace", request.Namespace, "Name", request.Name)
+			return reconcile.Result{}, err
+		}
+		if err := r.client.Create(context.TODO(), newSS); err != nil {
+			reqLogger.Error(err, "Error creating syncset", "Namespace", request.Namespace, "Name", request.Name)
+			return reconcile.Result{}, err
+		}
+		reqLogger.Info("Done creating a new SyncSet", "Namespace", request.Namespace, "Name", request.Name)
+
+	} else {
+		reqLogger.Info("SyncSet Already Present, nothing to do here...", "Namespace", request.Namespace, "Name", request.Name)
+		// return directly if the syscset already existed
+		return reconcile.Result{}, nil
+	}
+
+	// Check if the secret exists
+	err = r.client.Get(context.TODO(),
+		types.NamespacedName{Name: secretName, Namespace: request.Namespace},
+		&corev1.Secret{})
+
+	if errors.IsNotFound(err) {
+		// create new secret which will be referenced by SyncSet
+		reqLogger.Info("Secret not found, Creating a new Secret", "Namespace", request.Namespace, "Name", request.Name)
+
+		// snitchName := instance.Spec.ClusterName + "." + instance.Spec.BaseDomain
+		snitchName := "ds-test"
```

Comment:
```suggestion
		snitchName := instance.Spec.ClusterName + "." + instance.Spec.BaseDomain
```

Reply:
forgot to change it back which is used for local testing...